File: vuln-sandbox/python/taint/ssrf-1hop.py
# Taint: SSRF — 1-hop (Python)
# Source: request.args.get('url') → Sink: requests.get
# Expected: TruePositive (taint/cwe918)
from flask import Flask, request
import requests

app = Flask(__name__)

@app.route('/proxy')
def proxy():
    url = request.args.get('url')              # SOURCE: HTTP request param
    response = requests.get(url)               # SINK: outbound HTTP request
    return response.text

File: vuln-sandbox/node/taint/ssrf-1hop.js
// Taint: SSRF — 1-hop (JS)
// Source: req.query.url → Sink: fetch
// Expected: TruePositive (taint/cwe918)
const express = require('express');
const app = express();

app.get('/proxy', async (req, res) => {
    const url = req.query.url;                  // SOURCE: HTTP request param
    const response = await fetch(url);          // SINK: outbound HTTP request
    const data = await response.text();
    res.send(data);
});
